python/ray/data/_internal/dataset_logger.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import os
+from typing import Optional
+
+import ray
+from ray._private.ray_constants import LOGGER_FORMAT
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class DatasetLogger:
+    """Logger for Ray Datasets which, in addition to logging to stdout,
+    also writes to a separate log file at `DatasetLogger.DEFAULT_DATASET_LOG_PATH`.
+    """
+
+    DEFAULT_DATASET_LOG_PATH = "logs/datasets.log"
+
+    def __init__(self, log_name, dataset_log_path: Optional[str] = None):
```

Comment:
shall we introduce overriding `dataset_log_path` later if needed? We can always write to `DEFAULT_DATASET_LOG_PATH` for now.